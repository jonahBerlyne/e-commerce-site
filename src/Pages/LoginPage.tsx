import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { getAuth, signInWithEmailAndPassword } from "firebase/auth";
import "../Styles/Auth.css";

export default function LoginPage() {
 const [email, setEmail] = useState<string>('');
 const [password, setPassword] = useState<string>('');
 const auth = getAuth();

 const login = async (): Promise<any> => {
  try {
   await signInWithEmailAndPassword(auth, email, password);
  } catch (err) {
   alert(`Login error: ${err}`);
  }
 }

 return (
  <div>
   <h2>Login:</h2>
   <input 
    type="email" 
    className='form-control' placeholder='Email' 
    value={email} 
    onChange={(e) => {setEmail(e.target.value)}}
   />
   <input 
    type="password" 
    className='form-control' placeholder='Password' 
    value={password} 
    onChange={(e) => {setPassword(e.target.value)}}
   />
   <button className='my-3' onClick={login}>Login</button>
   <hr/>
   <Link to="/register">Click Here to Register</Link>
  </div>
 );
}