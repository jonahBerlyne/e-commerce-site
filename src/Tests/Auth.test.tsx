import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import LoginPage from "../Pages/LoginPage";

jest.mock("../firebaseConfig", () => {
  return {
    apps: ["testAppId"],
    auth: jest.fn(),
    firestore: jest.fn(),
  };
});

afterEach(cleanup);

describe("Login Page", () => {

 it("renders", () => {
  const { container } = render(<LoginPage />);
  expect(container).toMatchSnapshot();
 });

});