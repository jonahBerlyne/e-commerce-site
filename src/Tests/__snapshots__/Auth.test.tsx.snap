// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders login page 1`] = `
<div>
  <div
    class="auth"
  >
    <h2
      class="auth-header"
    >
      Login:
    </h2>
    <div
      class="auth-header-border"
    />
    <div
      class="auth-inputs"
    >
      <input
        class="form-control auth-input"
        data-testid="Email"
        placeholder="Email"
        type="email"
        value=""
      />
      <input
        class="form-control auth-input"
        data-testid="Password"
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      class="my-3 btn btn-primary auth-btn"
      disabled=""
    >
      Login
    </button>
    <hr />
    <a
      class="register-link"
      data-testid="register-link"
      href="/register"
    >
      Click Here to Register
    </a>
  </div>
</div>
`;

exports[`Register Page renders register page 1`] = `
<div>
  <div
    class="auth"
  >
    <h2
      class="auth-header"
    >
      Register:
    </h2>
    <div
      class="auth-header-border"
    />
    <div
      class="auth-inputs"
    >
      <input
        class="form-control auth-input"
        placeholder="Name"
        required=""
        type="text"
        value=""
      />
      <input
        class="form-control auth-input"
        data-testid="Email"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <input
        class="form-control auth-input"
        data-testid="Password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <input
        class="form-control auth-input"
        data-testid="confirmPassword"
        placeholder="Confirm Password"
        required=""
        type="password"
        value=""
      />
    </div>
    <button
      class="my-3 btn btn-primary auth-btn"
      disabled=""
    >
      Register
    </button>
    <a
      class="login-link"
      data-testid="login-link"
      href="/login"
    >
      Click Here to Login
    </a>
  </div>
</div>
`;
